patroni/ha.py

```diff
@@ -1695,15 +1692,18 @@ def _handle_dcs_error(self):
                 self.demote('offline')
                 return 'demoted self because DCS is not accessible and I was a leader'
             elif self.is_failsafe_mode():
-                cluster = self._failsafe.update_cluster(self.cluster)
-                if cluster:
-                    self.state_handler.slots_handler.sync_replication_slots(cluster,
-                                                                            self.patroni.nofailover,
-                                                                            self.patroni.replicatefrom,
-                                                                            self.is_paused())
-
+                self._sync_replication_slots()
         return 'DCS is not accessible'
 
+    def _sync_replication_slots(self):
```

Comment:
and the explanation how syncing of replication slots (=this function) works both for failsafe mode enabled and failing dcs (incl. this corner case of readonly dcs) and in the "normal operational mode"? (now, after an hour of despair, I think I got it)